packages/visual-editor/src/components/Dropzone/Hitboxes.tsx

```diff
@@ -0,0 +1,206 @@
+  }, [zoneId, fetchDomRect]);
+
+  const zones = useZoneStore((state) => state.zones);
+
+  const direction = zones[parentZoneId]?.direction || 'vertical';
+  const isVertical = direction === 'vertical';
+
+  const isRoot = parentZoneId === ROOT_ID;
+
+  const getStyles = useCallback(
+    (direction: HitboxDirection): CSSProperties => {
+      if (!domRect) {
+        return {
+          display: 'none',
+        };
+      }
+
+      const { width, height, top, left, bottom, right } = domRect;
+
+      const isDeepZone = zoneDepth > 5;
+      switch (direction) {
+        case HitboxDirection.TOP:
+          return {
+            width,
+            height: HITBOX_HEIGHT,
+            top: top - calcOffsetDepth(zoneDepth),
+            left,
+            zIndex: 100 + zoneDepth,
+          };
+        case HitboxDirection.BOTTOM:
+          return {
+            width,
+            height: HITBOX_HEIGHT,
+            top: bottom + calcOffsetDepth(zoneDepth),
+            left,
+            zIndex: 100 + zoneDepth,
+          };
+        case HitboxDirection.LEFT:
+          return {
+            width: HITBOX_WIDTH,
+            height: height - HITBOX_HEIGHT,
+            left: left - calcOffsetDepth(zoneDepth) - HITBOX_WIDTH / 2,
+            top: top + HITBOX_HEIGHT / 2,
+            zIndex: 100 + zoneDepth,
+          };
+        case HitboxDirection.RIGHT:
+          return {
+            width: HITBOX_WIDTH,
+            height: height - HITBOX_HEIGHT,
+            left: right - calcOffsetDepth(zoneDepth) - HITBOX_WIDTH / 2,
+            top: top + HITBOX_HEIGHT / 2,
+            zIndex: 100 + zoneDepth,
+          };
+        case HitboxDirection.SELF_VERTICAL: {
+          if (height > 60 && !isDeepZone) {
+            return { display: 'none' };
+          }
+
+          const selfHeight = isDeepZone ? MIN_DEPTH : MIN_HEIGHT;
+          return {
+            width,
+            height: selfHeight,
+            left,
+            top: top + height / 2 - selfHeight / 2,
+            zIndex: 1000 + zoneDepth,
+          };
+        }
+        case HitboxDirection.SELF_HORIZONTAL: {
+          if (width > DRAGGABLE_WIDTH) {
+            return { display: 'none' };
+          }
+          return {
+            width: width - DRAGGABLE_WIDTH * 2,
+            height,
+            left: left + (DRAGGABLE_WIDTH * 2) / 2,
+            top,
+            zIndex: 1000 + zoneDepth,
+          };
+        }
+        default:
+          return {};
+      }
+    },
+    [zoneDepth, domRect],
+  );
+
+  const ActiveHitboxes = (
+    <>
+      {isVertical ? (
+        <div
+          data-ctfl-zone-id={zoneId}
+          className={styles.hitbox}
+          style={getStyles(HitboxDirection.SELF_VERTICAL)}
+        />
+      ) : (
+        <div
+          data-ctfl-zone-id={zoneId}
+          className={styles.hitbox}
+          style={getStyles(HitboxDirection.SELF_HORIZONTAL)}
+        />
+      )}
```

Comment:
Some repetition could be avoided with a conditional value in the style prop
```suggestion
      <div
        data-ctfl-zone-id={zoneId}
        className={styles.hitbox}
        style={getStyles(isVertical ? HitboxDirection.SELF_VERTICAL : HitboxDirection.SELF_HORIZONTAL)}
      />
```